saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/provider/service/authentication/Saml2AuthenticationRequest.java

```diff
@@ -96,12 +104,38 @@ public String getAssertionConsumerServiceUrl() {
 		return this.credentials;
 	}
 
+	/**
+	 * Returns true if this AuthNRequest will be sent using REDIRECT binding and
+	 * encoded with the DEFLATE encoding. False if the AuthNRequest will be sent using POST binding.
+	 * @return true if REDIRECT/DEFLATE binding/encoding should be used. false for POST binding.
+	 */
+	public boolean isDeflate() {
```

Comment:
Referencing an offline discussion: Would it be better if this were about the binding chosen instead of whether or not to deflate?

Reply:
I agree, Please consider aeccdc562dd03a9a4d78d5e198144d92f469df0c as the solution